core/decoders/zlib.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * noVNC: HTML5 VNC client
+ * Copyright (C) 2024 The noVNC Authors
+ * Licensed under MPL 2.0 (see LICENSE.txt)
+ *
+ * See README.md for usage and integration instructions.
+ *
+ */
+
+import Inflator from "../inflator.js";
+
+export default class ZlibDecoder {
+    constructor() {
+        this._zlib = new Inflator();
+        this._length = 0;
+    }
+
+    decodeRect(x, y, width, height, sock, display, depth) {
+        if ((width === 0) || (height === 0)) {
+            return true;
+        }
+
+        const pixelSize = depth == 8 ? 1 : 4;
```

Comment:
The alternative depth is just for the raw encoding, so you can simplify things here and assume 32-bit.

Reply:
Fixed, removed from decoder and tests.